Python/scoreP3.py

```diff
@@ -93,47 +98,85 @@ def scoreP3(truthDir, testDir, phaseNum='3'):
 
-    # Build the Numpy arrays for calculations
-    truthValues = np.array([value['truth_value'] for value in combinedRows])
-    testValues = np.array([value['test_value'] for value in combinedRows])
-
-    # Compute accuracy, sensitivity, and specificity
-    truthBinaryValues = truthValues > 0.5
-    testBinaryValues = testValues > 0.5
-    metrics = computeCommonMetrics(truthBinaryValues, testBinaryValues)
-
-    # Compute average precision
-    metrics.extend(computeAveragePrecisionMetrics(truthValues, testValues))
-    
-    # Compute AUC
-    metrics.extend(computeAUCMetrics(truthValues, testValues))
-    
-    # Compute specificity at 95% sensitivity
-    metrics.extend(computeSPECMetrics(truthValues, testValues, 0.95))
-    
-    # Compute specificity at 98% sensitivity
-    metrics.extend(computeSPECMetrics(truthValues, testValues, 0.98))
-    
-    # Compute specificity at 99% sensitivity
-    metrics.extend(computeSPECMetrics(truthValues, testValues, 0.99))
+        metricsByCategory = {}
+    for category in ['melanoma', 'seborrheic_keratosis']:
+        metricsByCategory[category] = []
+
+        # Build the Numpy arrays for calculations
+        truthValues = np.array(
+            [value['truth_value_%s' % category] for value in combinedRows])
+        testValues = np.array(
+            [value['test_value_%s' % category] for value in combinedRows])
+
+        # Compute accuracy, sensitivity, and specificity
+        truthBinaryValues = truthValues > 0.5
+        testBinaryValues = testValues > 0.5
+        metricsByCategory[category].extend(
+            computeCommonMetrics(truthBinaryValues, testBinaryValues))
+
+        # Compute average precision
+        metricsByCategory[category].extend(
+            computeAveragePrecisionMetrics(truthValues, testValues))
+
+        # Compute specificity at 82% sensitivity
+        metricsByCategory[category].extend(
+            computeSPECMetrics(truthValues, testValues, 0.82))
+
+        # Compute specificity at 89% sensitivity
+        metricsByCategory[category].extend(
+            computeSPECMetrics(truthValues, testValues, 0.89))
+
+        # Compute specificity at 95% sensitivity
+        metricsByCategory[category].extend(
+            computeSPECMetrics(truthValues, testValues, 0.95))
+
+        # Compute AUC
+        metricsByCategory[category].extend(
+            computeAUCMetrics(truthValues, testValues))
+
+    # Compute mean metrics for combined melanoma and seborrheic_keratosis
+    metricsByCategory['mean'] = []
+    for metricMelanoma, metricSeborrheicKeratosis in zip(
+            metricsByCategory['melanoma'],
+            metricsByCategory['seborrheic_keratosis']):
+
+        assert metricMelanoma['name'] == metricSeborrheicKeratosis['name']
+        metricsByCategory['mean'].append({
+            'name': metricMelanoma['name'],
+            'value': (metricMelanoma['value'] +
+                      metricSeborrheicKeratosis['value']) / 2.0
+        })
```

Comment:
Even with 24 score columns, the Covalic submission score page renders in a sane way, albeit with a large horizontal scrollbar. If this was reduced to 16 columns there'd still be a large scrollbar, so it seems best to continue to compute the means here.